sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/raw_response.hpp>
+
+namespace Azure {
+
+template <class Derived> class PagedResponse {
+  bool m_hasMore = true;
```

Comment:
> nit: put under explicit private keyword

Done

Reply:
> Why is m_hasMore defaulted to true? Add a comment.

> I am wondering, why doesn't the initial call to ListX set this to true/false, rather than assuming m_hasMore is always true. What happens if ListX returns 0 items, do we still want to go through an iteration of the outer-loop?

> Maybe the derived classes OnNextPage call might be able to set this field. @JeffreyRichter what do you think?

I also find the logic here is kind of weird. As I see it, the user should call `NextPage()` before calling `HasMore()`, it basically works like:

```C++
auto pageResponse = ListSomething();
// handle this page
pageResponse.NextPage();
if (pageResponse.HasMore()) {
    // handle the second page
} else {
    // we're done here
}
```


It works, so I decided to keep it as-is, kind of counter-intuitive to me though.

Reply:
If I had to guess, the hasmore = true as the init state is to try to enable the for-loop pattern. If HasMore returned false as the first call, the loop would never run and never call NextPage.

`for (auto pr = clientAPI(); pr.HasMore(); pr.NextPage());`

**But,** if the clientAPI() which returns the PagedResponse already is making the first call to NextPage() for you, then we can let that method decide whether HasMore should be flipped to true. False as the default makes more sense to me, but curious to know what Jeff thinks.

Reply:
> If I had to guess, the hasmore = true as the init state is to try to enable the for-loop pattern. If HasMore returned false as the first call, the loop would never run and never call NextPage.

Yeah, I also think so.

But intuitively, `HasMore` should mean if there're more results on server side after the current page. But it actually means if there're more results on server side or in the current page.